integration-tests/trait-dyn-cross-contract-calls/lib.rs

```diff
@@ -0,0 +1,110 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// We can't run E2E test without the contract. So add `Dummy` contract.
+#[ink::contract]
+pub mod dummy {
+    #[ink(storage)]
+    #[derive(Default)]
+    pub struct Dummy {}
+
+    impl Dummy {
+        /// Dummy constructor.
+        #[ink(constructor)]
+        pub fn new() -> Self {
+            Default::default()
+        }
+
+        /// Dummy method.
+        #[ink(message)]
+        pub fn workaround(&self) {}
+    }
+}
+
+#[cfg(all(test, feature = "e2e-tests"))]
+mod e2e_tests {
+    use dyn_traits::Increment;
+    use ink::{
+        contract_ref,
+        env::DefaultEnvironment,
+    };
+    use ink_e2e::build_message;
+    use trait_incrementer::incrementer::IncrementerRef;
+    use trait_incrementer_caller::caller::CallerRef;
+
+    type E2EResult<T> = Result<T, Box<dyn std::error::Error>>;
+
+    #[ink_e2e::test(
+        additional_contracts = "contracts/incrementer/Cargo.toml contracts/incrementer-caller/Cargo.toml"
+    )]
+    async fn e2e_cross_contract_calls(
+        mut client: ink_e2e::Client<C, E>,
+    ) -> E2EResult<()> {
+        let _ = client
+            .upload("trait-incrementer", &ink_e2e::alice(), None)
+            .await
+            .expect("uploading `trait-incrementer` failed")
+            .code_hash;
+
+        let _ = client
+            .upload("trait-incrementer-caller", &ink_e2e::alice(), None)
+            .await
+            .expect("uploading `trait-incrementer-caller` failed")
+            .code_hash;
+
+        let constructor = IncrementerRef::new(
+            0, // initial value
+        );
+
+        let incrementer_account_id = client
+            .instantiate("trait-incrementer", &ink_e2e::alice(), constructor, 0, None)
+            .await
+            .expect("instantiate failed")
+            .account_id;
+
+        let constructor = CallerRef::new(incrementer_account_id.clone());
+
+        let caller_account_id = client
+            .instantiate(
+                "trait-incrementer-caller",
+                &ink_e2e::alice(),
+                constructor,
+                0,
+                None,
+            )
+            .await
+            .expect("instantiate failed")
+            .account_id;
+
+        // Check throw the caller that the value of the incrementer is zero
+        let get = build_message::<CallerRef>(caller_account_id.clone())
+            .call(|contract| contract.get());
+        let value = client
+            .call_dry_run(&ink_e2e::bob(), &get, 0, None)
```

Comment:
```suggestion
            .call_dry_run(&ink_e2e::alice(), &get, 0, None)
```

We can keep using `alice` throughout, unless you have a good reason not to